api/lib/application/users/index.js

```diff
@@ -73,9 +73,51 @@ exports.register = async function(server) {
           '- **Cette route est restreinte aux utilisateurs administrateurs**\n' +
           '- Elle permet de récupérer le détail d\'un utilisateur dans un contexte d\'administration\n',
         ],
-        tags: ['api', 'user admin'],
+        tags: ['api', 'administration' , 'user'],
       }
     },
+
+    {
+      method: 'PATCH',
+      path: '/api/admin/users/{id}',
+      config: {
+        pre: [{
+          method: securityPreHandlers.checkUserHasRolePixMaster,
+          assign: 'hasRolePixMaster'
+        }],
+        plugins: {
+          'hapi-swagger': {
+            payloadType: 'form'
+          }
+        },
+        validate: {
+          params: Joi.object({
+            id: Joi.number().integer().required(),
+          }),
+          payload: Joi.object({
+            data: {
+              attributes: {
+                'first-name': Joi.string().empty(Joi.string().regex(/^\s*$/)).required(),
+                'last-name': Joi.string().empty(Joi.string().regex(/^\s*$/)).required(),
+                email: Joi.string().email().allow(null).optional(),
+              }
+            }
+          }),
+          options: {
+            allowUnknown: true
+          },
+          failAction: (request, h , err) => {
+            throw err;
```

Comment:
Dans l'index de authentication, il y a une gestion explicite avec une remontée de 400, qu'en penses-tu ? Est-ce applicable ici ?
```
          failAction: (request, h) => {
            const errorHttpStatusCode = 400;
            const jsonApiError = new JSONAPIError({
              code: errorHttpStatusCode.toString(),
              title: 'Bad request',
              detail: 'Les données envoyées ne sont pas au bon format.',
            });
            return h.response(jsonApiError).code(errorHttpStatusCode).takeover();
          }
```

Reply:
C'est que j'avais juste avant, je l'ai changé car j'avais besoin de plus de détail sur l'erreur dans mes tests pour le bad request. 
Par exemple expect(firstError.detail[0].message).to.equal('"data.attributes.first-name" is required');
J'avais fait un PR pour essayer de regrouper cela au niveau serveur https://github.com/1024pix/pix/pull/925/commits/466e6fc7a5d8629ff3c1ffaf4d4594eff7bcafd1
Néanmoins, je me dis qu'il faut garder la même structure de l'objet error même si on l'interprète pas coté front. Je vais modifier pour garder la même structure même avec une erreur dynamique ce qui me semble plus cohérent.

`          failAction: (request, h , err) => {
            const errorHttpStatusCode = 400;
            const jsonApiError = new JSONAPIError({
              code: errorHttpStatusCode.toString(),
              title: 'Bad request',
              detail: err.details,
            });
            return h.response(jsonApiError).code(errorHttpStatusCode).takeover();
          }`